### DISCUSSION TITLE
Access remote AWS Oracle server on Windows DBeaver client via Unix server

### DISCUSSION THREAD

**Discussion by sasikpkw** [dbeaver/dbeaver#10576] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version
Windows Server 2012 R2 Standard
- DBeaver version
7.0.1

#### Connection specification:
- Database name and version
Oracle 12c

#### Describe your question:
We are able to access the Oracle 12c server (say with hostname, ora.domain.com, port 1234) from a unix box (say with hostname as rhel.domain.com) - the firewalls between these 2 servers are cleared appropriately. The Windows machine we use is not cleared to access the Oracle server and also has no internet connection (resides behind a corporate firewall). The Windows machine is able to access the unix box (rhel.domain.com) on port 22. So we tried to 
1. Use manually placed Oracle drivers and remove the default Apache Maven drivers (which we understand need internet access). Picked the drivers from here:
https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html
2. use SSH tunneling through the unix box

I am getting the below error:
IO Error: The Network Adapter could not establish the connection
    Connection timed out: connect, socket connect lapse 21031 ms.
![image](https://user-images.githubusercontent.com/75334511/100788796-960bc280-343b-11eb-98dd-9167e2551d83.png)

I tried increasing the timeout to 60s but it was still timing out at ~20s. I see no lag while connecting from Windows to the unix box or from the unix box to the Oracle servers (all the tns entries respond within 40ms to tnspings).
I am wondering if am doing something wrong with the SSH tunneling.

Please verify If you can suggest me to try something. Since I am querying reg a customer system, I have put masked/dummy names and masked image. If you need some more detailed inputs (with better screenshots) I will try to get them.

**Comment by sasikpkw** [comment]
I was able to test the SSH tunnel connection to be successful (I put in the hostname and credentials to the unix box from where we typically access the Oracle server):
![image](https://user-images.githubusercontent.com/75334511/100789926-226ab500-343d-11eb-8d3e-16baa304406f.png)

**Comment by p91paul** [comment]
Are you using an ip address or an hostname? It's possible that your local machine cannot resolve the remote database IP. You should try using the IP that is seen by the tunneling machine.

**Comment by sasikpkw** [comment]
@p91paul I gave the same entries which the tunnel host is able to resolve. I also tried giving the IP of the Oracle DB in DBeaver as per your suggestion - getting same error.

**Comment by sasikpkw** [comment]
Hi, looking for some guidance here. To add, I have made entries in the Windows host file, to help resolve the hostnames as well.

**Comment by kseniiaguzeeva** [comment, marked as answer]
Please try to

1. Create a new database connection with default driver settings (change nothing there)
2. Go to SSH tab and fill necessary fields
3. Press **Test tunnel configuration** (If the test fails, send us log files https://github.com/dbeaver/dbeaver/wiki/Log-files)
4. Fill other parameters in the Main tab of Connection settings and press **Test connection**

**Comment by sasikpkw** [comment]
@kseniiaguzeeva Thanks for the suggestion. Closing all the sessions, and altering the sequence (drivers, SSH and Test Connection) was able to make things work - not really sure if it is this sequence or random system states that made things work. 
I am closing the ticket.
